app/Models/WaitingRoom.php

```diff
@@ -88,11 +88,15 @@ public function getDefaultSplit()
 
     /*
      * Creates competitions based on the given user split.
+     *
+     * @param \Gladiator\Models\Contest $contest
+     * @param Array                     $split
+     * @param \Illuminate\Http\Request  $request
      */
-    public function saveSplit($contest, $split, $endDate)
+    public function saveSplit($contest, $split, $request)
     {
         $startDate = Carbon::now()->startOfDay();
-        $endDate = new Carbon($endDate);
+        $endDate = new Carbon($request->competition_end_date);
```

Comment:
@DFurnes if so, couldn't you just cast what you pass to be a simple object? unless that's like a bad practice or something :confused: